build: add more shell scripts to the `lintsh` target

dundargoc opened the pull request:
- build: add EXCLUDE option to add_glob_target

EXCLUDE filters out all elements containing regex, meaning it works on
both files and directories.

Also rename add_glob_targets to add_glob_target since only one target is
being created.

- build(lint): add more shell scripts to lintsh

dundargoc commented:
Alternative solution could be to remove `lintsh` and to reduce shell usage.

dundargoc commented:
I'm starting to suspect this may not be worth the effort. Closing for now.

justinmk commented:
For shell scripts that aren't going away it's definitely wanted.

These scripts will be deleted pretty soon: `bump-deps.sh` and `lua2dox_filter`

dundargoc committed:
build(lint): add more shell scripts to lintsh

dundargoc commented:
This is pretty much done. `genappimage` isn't our script and `pvscheck` feels not worth messing with.

justinmk commented:
> * disable SC2086
> 
> It's noisy and often times just wrong. Not worth it.

Don't think we should disable that. Yes, we have scripts where we don't use quotes because we "know" that the variable doesn't have whitespace, but almost all of these cases can easily be updated to be quoted. The few cases where we don't want double quotes are not worth disabling this important check.

justinmk commented:
> This is pretty much done. `genappimage` isn't our script and `pvscheck` feels not worth messing with.

we own genappimage. can wait (indefinitely) on pvscheck if it's too much work.

justinmk commented on CMakeLists.txt:
```diff
@@ -638,7 +638,16 @@ include(InstallHelpers)
+    ci/before_cache.sh
+    ci/before_script.sh
+    ci/install.sh
+    ci/run_tests.sh
```
can we make this an exclude instead of include? or we still have too many unchecked scripts?

dundargoc replied:
Yeah, it's not pretty. I'll come up with something better.

dundargoc commented:
> we own genappimage.

Oh. Then there's no reason not to glob lint everything in `scripts` directory then.

dundargoc committed:
build: add EXCLUDE option to add_glob_target

EXCLUDE filters out all elements containing regex, meaning it works on
both files and directories.

Also rename add_glob_targets to add_glob_target since only one target is
being created.

dundargoc commented:
Added exclude option to `add_glob_target` as it's easier to just skip `pvscheck.sh` rather than trying to decipher the madness.

dundargoc commented:
@justinmk please take a look again, this should be a cleaner solution.

justinmk commented on scripts/genappimage.sh:
```diff
@@ -34,8 +35,9 @@ make install
+export VERSION
 
-cd "$APP_BUILD_DIR"
+cd "$APP_BUILD_DIR" || exit
```
should we exit with error code here? same for the other case below

dundargoc replied:
It already does I think, `cd A || exit` is the same as `cd A || exit 1` as I understand it. Although it doesn't really hurt to be explicit so we might as well use `exit 1` for readability.

justinmk replied:
oh I should have read `exit -help` : 

> Exits the shell with a status of N.  If N is omitted, the exit status is that of the last command executed.

That's definitely better. So it's good as you have it.

justinmk commented on scripts/git-log-pretty-since.sh:
```diff
@@ -40,13 +40,13 @@ _format_ticketnums() {
-  if is_merge_commit ${commit} ; then
-      if [ -z "$__INVMATCH" ] || ! git log --oneline ${commit}^1..${commit}^2 \
+  if is_merge_commit "${commit}" ; then
+      if [ -z "$__INVMATCH" ] || ! git log --oneline "${commit}"^1.."${commit}"^2 \
```
nit (similar cases below): 

```suggestion
      if [ -z "$__INVMATCH" ] || ! git log --oneline "${commit}^1..${commit}^2" \
```

justinmk commented on scripts/release.sh:
```diff
@@ -59,8 +59,8 @@ _do_release_commit() {
-    build/bin/nvim --api-info > test/functional/fixtures/api_level_$__API_LEVEL.mpack
-    git add test/functional/fixtures/api_level_$__API_LEVEL.mpack
+    build/bin/nvim --api-info > test/functional/fixtures/api_level_"$__API_LEVEL".mpack
+    git add test/functional/fixtures/api_level_"$__API_LEVEL".mpack
```
is this more readable / less error-prone? 

```suggestion
    git add "test/functional/fixtures/api_level_${__API_LEVEL}.mpack"
```

justinmk reviewed (approved):
nice!

justinmk reviewed (approved):
LGTM, I'd squash most of the script fixups
